impl/src/main/java/io/kroxylicious/testing/kafka/testcontainers/TestcontainersKafkaCluster.java

```diff
@@ -276,13 +277,34 @@ private synchronized String buildBrokerList(Function<Integer, KafkaClusterConfig
                 .collect(Collectors.joining(","));
     }
 
+    /**
+     * Gets the container image version tag for the specified kafka version.
+     * This knows about the naming conventions used by the ogunalp native image project.
+     *
+     * @param kafkaVersion desired kafka version, which may be null.
+     * @return image tag or null
+     */
+    @Nullable
+    private String getContainerImageVersionTag(String kafkaVersion) {
+        if (kafkaVersion == null) {
+            return null;
+        }
+        else {
+            return kafkaVersion.equals("latest") ? kafkaVersion : "latest-kafka-" + kafkaVersion;
+        }
+    }
+
     /**
      * Gets kafka version.
      *
      * @return the kafka version
      */
     public String getKafkaVersion() {
-        return kafkaImage.getVersionPart();
+        var v = kafkaImage.getVersionPart();
+        if (v != null) {
+            v = v.replaceFirst("^latest-kafka-", "");
```

Comment:
I've kept the `latest-kafka-` prefix a private concern of the implementation.  Technically this changes the `getKafkaVersion` API, but if you squint at it, you can convince yourself it was a bug all along.